tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/RunMojo.java

```diff
@@ -145,12 +185,30 @@ private void deploy(BundleContext bundleContext) throws MojoExecutionException {
                 } else {
                     throw new MojoExecutionException("Packaging " + project.getPackaging() + " is not supported");
                 }
+            } else if (attachedFeatureFileExists) {
+                addFeaturesAttachmentAsFeatureRepository(featureService, attachedFeatureFile);
             } else {
                 throw new MojoExecutionException("Project artifact doesn't exist");
             }
         }
     }
 
+    void addFeatures(Object featureService) throws MojoExecutionException {
+    	if (featuresToInstall != null) {
+            try {
+            	Class<? extends Object> serviceClass = featureService.getClass();
+                Method installFeatureMethod = serviceClass.getMethod("installFeature", String.class);
+                String[] features = featuresToInstall.split(" *, *");
+                for (String feature : features) {
+                    installFeatureMethod.invoke(featureService, feature);
+                    Thread.sleep(1000L);
```

Comment:
I used my own application to test it and my own application triggers [KARAF-5174](https://issues.apache.org/jira/browse/KARAF-5174). In addition to the uninstall problem there is a startup problem seen in the karaf.log if the features are loaded together (the derby driver fails to be able to create the memory database), but if the liquibase-core feature is installed _before_ the database feature there are no such issues in the karaf.log.

I saw something similar in the karaf.log when running karaf:run on my app, so I tried the 1s sleep to simulate what I do manually at the karaf console.  But it didn't make any difference.  So feel free to drop the sleep if it doesn't make sense.